servers/src/main/java/tachyon/worker/block/PinListSync.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.master.MasterClient;
+import tachyon.util.CommonUtils;
+import tachyon.util.NetworkUtils;
+import tachyon.util.ThreadFactoryUtils;
+
+/**
+ * Task that fetches pinlist from master periodically
+ *
+ */
+public class PinListSync implements Runnable {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  /** Block data manager responsible for interacting with Tachyon and UFS storage */
+  private final BlockDataManager mBlockDataManager;
+  /** The executor service for the master client thread */
+  private final ExecutorService mMasterClientExecutorService;
+  /** The configuration values */
+  private final TachyonConf mTachyonConf;
+  /** Milliseconds between each fetch */
+  private final int mFetchIntervalMs;
+  /** Milliseconds between fetches before a timeout */
+  private final int mFetchTimeoutMs;
```

Comment:
Maybe rename `fetch` to `sync` (or vice versa)? Otherwise there are too many similar terms being used.